stdlib/public/core/CollectionAlgorithms.swift.gyb

```diff
@@ -141,6 +144,7 @@ extension MutableCollectionType
 ${partitionDocComment}
   ///
 ${orderingRequirementForComparable}
+  @swift3_migration(message="Slice the collection using the range, and call partition()")
   public mutating func partition(range: Range<Index>) -> Index {
```

Comment:
How about: Slice the collection using the range, and call partition(by:) with an appropriate closure.